src/pages/Transfer/TransferForm/index.tsx

```diff
@@ -0,0 +1,220 @@
+import isValidPolkadotAddress from 'utils/validation/validate-address';
+import STATUSES from 'utils/constants/statuses';
+import {
+  KUSAMA,
+  POLKADOT
+} from 'utils/constants/relay-chain-names';
+
+const TRANSFER_AMOUNT = 'wrapped-token-input-amount';
+const RECIPIENT_ADDRESS = 'collateral-token-address';
+
+type TransferFormData = {
+  [TRANSFER_AMOUNT]: string;
+  [RECIPIENT_ADDRESS]: string;
+}
+
+const TransferForm = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+
+  const {
+    parachainStatus,
+    address
+  } = useSelector((state: StoreType) => state.general);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<TransferFormData>({
+    mode: 'onChange'
+  });
+
+  const [activeToken, setActiveToken] = React.useState<TokenOption | undefined>(undefined);
+  const [accountSet, setAccountSet] = React.useState<boolean | undefined>(undefined);
+  const [submitStatus, setSubmitStatus] = React.useState(STATUSES.IDLE);
+  const [submitError, setSubmitError] = React.useState<Error | null>(null);
+
+  const onSubmit = async (data: TransferFormData) => {
+    if (!activeToken) return;
+
+    try {
+      setSubmitStatus(STATUSES.PENDING);
+
+      await window.bridge.interBtcApi.tokens.transfer(
+        data[RECIPIENT_ADDRESS],
+        newMonetaryAmount(data[TRANSFER_AMOUNT], activeToken.token as Currency<CurrencyUnit>, true)
+      );
+
+      setSubmitStatus(STATUSES.RESOLVED);
+    } catch (error: any) {
+      setSubmitStatus(STATUSES.REJECTED);
+      setSubmitError(error);
+    }
+  };
+
+  const validateTransferAmount = React.useCallback((value: number): string | undefined => {
+    if (!activeToken) return;
+
+    const balance = newMonetaryAmount(activeToken.balance, activeToken.token as Currency<CurrencyUnit>, true);
+    const transferAmount = newMonetaryAmount(value, activeToken.token as Currency<CurrencyUnit>, true);
+
+    return balance.lt(transferAmount) ? t('insufficient_funds') : undefined;
+  }, [activeToken, t]);
+
+  const validateAddress = React.useCallback((address: string): string | undefined => {
+    return isValidPolkadotAddress(address) ? undefined : t('validation.invalid_polkadot_address');
+  }, [t]);
+
+  const handleConfirmClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!accountSet) {
+      dispatch(showAccountModalAction(true));
+      event.preventDefault();
+    }
+  };
+
+  const handleTokenChange = (token: any) => {
+    setActiveToken(token);
+  };
+
+  React.useEffect(() => {
+    setAccountSet(!!address);
+  }, [address]);
+
+  // This ensures that triggering the notification and clearing
+  // the form happen at the same time.
+  React.useEffect(() => {
+    if (submitStatus !== STATUSES.RESOLVED) return;
+
+    toast.success(t('transfer_page.successfully_transferred'));
+
+    reset({
+      [TRANSFER_AMOUNT]: '',
+      [RECIPIENT_ADDRESS]: ''
+    });
```

Comment:
@tomjeatt 
I think this part can go into below line 77 instead of this React.useEffect hook.